lib/TickLib.sol

```diff
@@ -5,26 +5,52 @@ import {Bin} from "mgv_lib/BinLib.sol";
 import "mgv_lib/Constants.sol";
 import "mgv_lib/TickConversionLib.sol";
 
+type Tick is int;
+using TickLib for Tick global;
+
 library TickLib {
 
-  function inRange(int tick) internal pure returns (bool) {
-    return tick >= MIN_TICK && tick <= MAX_TICK;
+  function inRange(Tick tick) internal pure returns (bool) {
+    return Tick.unwrap(tick) >= MIN_TICK && Tick.unwrap(tick) <= MAX_TICK;
+  }
+
+  function eq(Tick tick1, Tick tick2) internal pure returns (bool) {
+    unchecked {
+      return Tick.unwrap(tick1) == Tick.unwrap(tick2);
+    }
   }
-  function fromBin(Bin bin, uint tickSpacing) internal pure returns (int) {
-    return Bin.unwrap(bin) * int(tickSpacing);
+
+  // Returns the nearest, higher bin to the given tick at the given tickSpacing
+  function toNearestBin(Tick tick, uint tickSpacing) internal pure returns (Bin) {
+    unchecked {
+      // Do not force ticks to fit the tickSpacing (aka tick%tickSpacing==0)
+      // Round maker ratios up such that maker is always paid at least what they asked for
+      int bin = Tick.unwrap(tick) / int(tickSpacing);
+      if (Tick.unwrap(tick) > 0 && Tick.unwrap(tick) % int(tickSpacing) != 0) {
+        bin = bin + 1;
+      }
+      return Bin.wrap(bin);
+    }
+  }
+
+
+  // Optimized conversion for ticks that are known to map exactly to a bin at the given tickSpacing,
+  // eg for offers in the offer list which are always written with a tick-aligned tick
+  function alignedToNearestBin(Tick tick, uint tickSpacing) internal pure returns (Bin) {
```

Comment:
As noted elsewhere, this reads weird when used: `tick.alignedToNearestBin(tickSpacing)`. Also, since the `tick` must be aligned with a bin, the conversion is exact and `nearest` is misleading.

Maybe `tick.toAlignedBin()`?

Reply:
removed in 6232ec08cfa33cb5f609747d6962a9f289e3d775